### PR TITLE
Semi-working-ish system

### PR DESCRIPTION
**Pull request by norling**
Well - this isn't exactly pretty, but it's the code that's currently running ^^

It was mostly designed to have something working quickly, then I spent about a day trying to clean things up, but it's still very messy. Dunno if it's better to work more on it, or if we should merge this and then start cleaning it up?

### CODE REVIEW

**Review by shreyasshivakumara — APPROVED** [review]
Haven't gone through the code yet. I tried implementing and It works! :)

File: src/software_router.py [hunk 1]
```diff
@@ -3,41 +3,404 @@
...
 from json.decoder import JSONDecodeError
 
+from settings import Settings
+from repository import Repository
 
-def load_settings_file(filename):
+def kube_label(name: str) -> str:
     """
-    Returns a settings dictionary from the data in `filename` or None
+    kubernetes job names must bu only lower case alphanumeric characters, or
```
Comment (dbampalikis) [inline src/software_router.py:38]:
```suggestion
    kubernetes job names must be only lower case alphanumeric characters, or
```

File: src/settings.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                parse_to_namespaces(value, attribute)
+            else:
+                setattr(target, key, value)
+
+
+class Settings():
+
+    def __init__(self, filename: str = "settings.yaml"):
+        """
+        Sets default values for all settings, lthen loads the given settings
```
Comment (dbampalikis) [inline src/settings.py:34]:
```suggestion
        Sets default values for all settings, then loads the given settings
```

**Review by dbampalikis — APPROVED** [review]
After an awesome explanation in pair review with Martin, I am convinced this is ready to be merged! Good job in general, seems complicated to deal with it!

**Review by jbygdell — APPROVED** [review]

**Review by aaperis — APPROVED** [review]
good job!  I am not sure about the logic behind the repository.py, but it seems to work well. Some documentation may be helpful in the future :-)

**Review by kostas-kou — APPROVED** [review]

**Pull request merged**